volmdlr/faces.py

```diff
@@ -5568,16 +5570,29 @@ def pair_with(self, other_bspline_face3d):
         corresponding_direction
         grid2d_direction
         '''
-        
-        adjacent_direction1, diff1, adjacent_direction2, diff2 = self.adjacent_direction(other_bspline_face3d)
-        start1, end1, start2, end2 = self.extremities(other_bspline_face3d)
 
+        adjacent_direction1, diff1, adjacent_direction2, diff2 = self.adjacent_direction(other_bspline_face3d)
         corresponding_directions = []
         if (diff1 > 0 and diff2 > 0) or (diff1 < 0 and diff2 < 0):
             corresponding_directions.append(('+' + adjacent_direction1, '+' + adjacent_direction2))
         else:
             corresponding_directions.append(('+' + adjacent_direction1, '-' + adjacent_direction2))
 
+        if adjacent_direction1 == 'u' and adjacent_direction2 == 'u':
+            corresponding_directions, grid2d_direction = self.adjacent_direction_uu(other_bspline_face3d, corresponding_directions)
+        elif adjacent_direction1 == 'v' and adjacent_direction2 == 'v':
+            corresponding_directions, grid2d_direction = self.adjacent_direction_vv(other_bspline_face3d, corresponding_directions)
+        elif adjacent_direction1 == 'u' and adjacent_direction2 == 'v':
+            corresponding_directions, grid2d_direction = self.adjacent_direction_uv(other_bspline_face3d, corresponding_directions)
+        elif adjacent_direction1 == 'v' and adjacent_direction2 == 'u':
+            corresponding_directions, grid2d_direction = self.adjacent_direction_vu(other_bspline_face3d, corresponding_directions)
```

Comment:
FYI: you can use the `getattr` keyword here.
```suggestion
        corresponding_directions, grid2d_direction = getattr(self, 'adjacent_direction_' + adjacent_direction1 + adjacent_direction2)(other_bspline_face3d, corresponding_directions)
```